# Le dataset

Dans ce challenge on utilisera le dataset ci-après [dataset food](https://drive.google.com/open?id=1L4tDDnZWwPDjeRKlleoYHgic2_xNhZ8i). Attention il est volumineux !!!

Tu l'utiliseras pour mener une analyse descriptive et exploratoire. L'objectif est d'arriver à un dataset permettant de répondre à la question : "Est-ce que cet aliment est sain ou non ?"



# Mission 1


Trouve le moyen d'ouvrir le fichier et d'afficher les 10 premiers éléments du dataset ci-dessus.

Ensuite affiche le nombre de colonne objet et float.

In [1]:
import pandas as pd
from pydrive.auth import GoogleAuth
from pydrive.drive import GoogleDrive
from google.colab import auth
from oauth2client.client import GoogleCredentials

# Authenticate and create the PyDrive client.
# The goal is to connect your Google Drive with this Google colab notebook.
# This only needs to be done once per notebook.

auth.authenticate_user()
gauth = GoogleAuth()
gauth.credentials = GoogleCredentials.get_application_default()
drive = GoogleDrive(gauth)
# This part is here to copy the file into your Drive (because we can't open a shared file)
url = 'https://drive.google.com/file/d/1L4tDDnZWwPDjeRKlleoYHgic2_xNhZ8i/view'
file_id = '1L4tDDnZWwPDjeRKlleoYHgic2_xNhZ8i'
downloaded = drive.CreateFile({'id': file_id})
downloaded.GetContentFile('fr.openfoodfacts.org.products.csv')
df_yuka = pd.read_csv('fr.openfoodfacts.org.products.csv', sep='\t')

<ipython-input-1-0dd708971972>:20: DtypeWarning: Columns (0,3,5,19,20,24,25,26,27,28,35,36,37,38,39,48) have mixed types. Specify dtype option on import or set low_memory=False.
  df_yuka = pd.read_csv('fr.openfoodfacts.org.products.csv', sep='\t')


In [3]:
df_yuka.head(10)

code                                                url  \
0   3087  http://world-fr.openfoodfacts.org/produit/0000...   
1   4530  http://world-fr.openfoodfacts.org/produit/0000...   
2   4559  http://world-fr.openfoodfacts.org/produit/0000...   
3  16087  http://world-fr.openfoodfacts.org/produit/0000...   
4  16094  http://world-fr.openfoodfacts.org/produit/0000...   
5  16100  http://world-fr.openfoodfacts.org/produit/0000...   
6  16117  http://world-fr.openfoodfacts.org/produit/0000...   
7  16124  http://world-fr.openfoodfacts.org/produit/0000...   
8  16193  http://world-fr.openfoodfacts.org/produit/0000...   
9  16513  http://world-fr.openfoodfacts.org/produit/0000...   

                      creator   created_t      created_datetime  \
0  openfoodfacts-contributors  1474103866  2016-09-17T09:17:46Z   
1             usda-ndb-import  1489069957  2017-03-09T14:32:37Z   
2             usda-ndb-import  1489069957  2017-03-09T14:32:37Z   
3             usda-ndb-import  1489055731  2017-03-09T10:35:31Z   
4             usda-ndb-import  1489055653  2017-03-09T10:34:13Z   
5             usda-ndb-import  1489055651  2017-03-09T10:34:11Z   
6             usda-ndb-import  1489055730  2017-03-09T10:35:30Z   
7             usda-ndb-import  1489055711  2017-03-09T10:35:11Z   
8             usda-ndb-import  1489055651  2017-03-09T10:34:11Z   
9             usda-ndb-import  1489055654  2017-03-09T10:34:14Z   

  last_modified_t last_modified_datetime                       product_name  \
0      1474103893   2016-09-17T09:18:13Z                 Farine de blé noir   
1      1489069957   2017-03-09T14:32:37Z     Banana Chips Sweetened (Whole)   
2      1489069957   2017-03-09T14:32:37Z                            Peanuts   
3      1489055731   2017-03-09T10:35:31Z             Organic Salted Nut Mix   
4      1489055653   2017-03-09T10:34:13Z                    Organic Polenta   
5      1489055651   2017-03-09T10:34:11Z  Breadshop Honey Gone Nuts Granola   
6      1489055730   2017-03-09T10:35:30Z      Organic Long Grain White Rice   
7      1489055712   2017-03-09T10:35:12Z                     Organic Muesli   
8      1489055651   2017-03-09T10:34:11Z       Organic Dark Chocolate Minis   
9      1489055654   2017-03-09T10:34:14Z              Organic Sunflower Oil   

  generic_name quantity  ... ph_100g fruits-vegetables-nuts_100g  \
0          NaN      1kg  ...     NaN                         NaN   
1          NaN      NaN  ...     NaN                         NaN   
2          NaN      NaN  ...     NaN                         NaN   
3          NaN      NaN  ...     NaN                         NaN   
4          NaN      NaN  ...     NaN                         NaN   
5          NaN      NaN  ...     NaN                         NaN   
6          NaN      NaN  ...     NaN                         NaN   
7          NaN      NaN  ...     NaN                         NaN   
8          NaN      NaN  ...     NaN                         NaN   
9          NaN      NaN  ...     NaN                         NaN   

  collagen-meat-protein-ratio_100g cocoa_100g chlorophyl_100g  \
0                              NaN        NaN             NaN   
1                              NaN        NaN             NaN   
2                              NaN        NaN             NaN   
3                              NaN        NaN             NaN   
4                              NaN        NaN             NaN   
5                              NaN        NaN             NaN   
6                              NaN        NaN             NaN   
7                              NaN        NaN             NaN   
8                              NaN        NaN             NaN   
9                              NaN        NaN             NaN   

  carbon-footprint_100g nutrition-score-fr_100g nutrition-score-uk_100g  \
0                   NaN                     NaN                     NaN   
1                   NaN                    14.0                    14.0   
2           

In [141]:
df_yuka.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 320772 entries, 0 to 320771
Columns: 162 entries, code to water-hardness_100g
dtypes: float64(106), object(56)
memory usage: 396.5+ MB


- 56 colonnes de type 'object'
- 106 colonnes de type 'float'

In [3]:
df_yuka.describe()

no_nutriments    additives_n  ingredients_from_palm_oil_n  \
count            0.0  248939.000000                248939.000000   
mean             NaN       1.936024                     0.019659   
std              NaN       2.502019                     0.140524   
min              NaN       0.000000                     0.000000   
25%              NaN       0.000000                     0.000000   
50%              NaN       1.000000                     0.000000   
75%              NaN       3.000000                     0.000000   
max              NaN      31.000000                     2.000000   

       ingredients_from_palm_oil  ingredients_that_may_be_from_palm_oil_n  \
count                        0.0                            248939.000000   
mean                         NaN                                 0.055246   
std                          NaN                                 0.269207   
min                          NaN                                 0.000000   
25%                          NaN                                 0.000000   
50%                          NaN                                 0.000000   
75%                          NaN                                 0.000000   
max                          NaN                                 6.000000   

       ingredients_that_may_be_from_palm_oil  nutrition_grade_uk  \
count                                    0.0                 0.0   
mean                                     NaN                 NaN   
std                                      NaN                 NaN   
min                                      NaN                 NaN   
25%                                      NaN                 NaN   
50%                                      NaN                 NaN   
75%                                      NaN                 NaN   
max                                      NaN                 NaN   

        energy_100g  energy-from-fat_100g       fat_100g  ...    ph_100g  \
count  2.611130e+05            857.000000  243891.000000  ...  49.000000   
mean   1.141915e+03            585.501214      12.730379  ...   6.425698   
std    6.447154e+03            712.809943      17.578747  ...   2.047841   
min    0.000000e+00              0.000000       0.000000  ...   0.000000   
25%    3.770000e+02             49.400000       0.000000  ...   6.300000   
50%    1.100000e+03            300.000000       5.000000  ...   7.200000   
75%    1.674000e+03            898.000000      20.000000  ...   7.400000   
max    3.251373e+06           3830.000000     714.290000  ...   8.400000   

       fruits-vegetables-nuts_100g  collagen-meat-protein-ratio_100g  \
count                  3036.000000                        165.000000   
mean                     31.458587                         15.412121   
std                      31.967918                          3.753028   
min                       0.000000                          8.000000   
25%                       0.000000                         12.000000   
50%                      23.000000                         15.000000   
75%                      51.000000                         15.000000   
max                     100.000000                         25.000000   

       cocoa_100g  chlorophyl_100g  carbon-footprint_100g  \
count  948.000000              0.0             268.000000   
mean    49.547785              NaN             341.700764   
std     18.757932              NaN             425.211439   
min      6.000000              NaN               0.000000   
25%     32.000000              NaN              98.750000   
50%     50.000000              NaN             195.750000   
75%     64.250000              NaN             383.200000   
max    100.000000              NaN            2842.000000   

       nutrition-score-fr_100g  nutrition-score-uk_100g  glycemic-index_100g  \
count            221210.000000            221210.000000                  0.0   
mean                  9.165535             

In [4]:
df_yuka.isna().sum()

code                           23
url                            23
creator                         2
created_t                       3
created_datetime                9
                            ...  
carbon-footprint_100g      320504
nutrition-score-fr_100g     99562
nutrition-score-uk_100g     99562
glycemic-index_100g        320772
water-hardness_100g        320772
Length: 162, dtype: int64

# Mission 2

Trouve le moyen d'afficher d'une part **uniquement** les noms des colonnes objet, et d'autre part **uniquement** les noms des colonnes float.

In [4]:
df_yuka.select_dtypes('object').columns

Index(['code', 'url', 'creator', 'created_t', 'created_datetime',
       'last_modified_t', 'last_modified_datetime', 'product_name',
       'generic_name', 'quantity', 'packaging', 'packaging_tags', 'brands',
       'brands_tags', 'categories', 'categories_tags', 'categories_fr',
       'origins', 'origins_tags', 'manufacturing_places',
       'manufacturing_places_tags', 'labels', 'labels_tags', 'labels_fr',
       'emb_codes', 'emb_codes_tags', 'first_packaging_code_geo', 'cities',
       'cities_tags', 'purchase_places', 'stores', 'countries',
       'countries_tags', 'countries_fr', 'ingredients_text', 'allergens',
       'allergens_fr', 'traces', 'traces_tags', 'traces_fr', 'serving_size',
       'additives', 'additives_tags', 'additives_fr',
       'ingredients_from_palm_oil_tags',
       'ingredients_that_may_be_from_palm_oil_tags', 'nutrition_grade_fr',
       'pnns_groups_1', 'pnns_groups_2', 'states', 'states_tags', 'states_fr',
       'main_category', 'main_category_fr', 'i

In [5]:
df_yuka.select_dtypes('float').columns

Index(['no_nutriments', 'additives_n', 'ingredients_from_palm_oil_n',
       'ingredients_from_palm_oil', 'ingredients_that_may_be_from_palm_oil_n',
       'ingredients_that_may_be_from_palm_oil', 'nutrition_grade_uk',
       'energy_100g', 'energy-from-fat_100g', 'fat_100g',
       ...
       'ph_100g', 'fruits-vegetables-nuts_100g',
       'collagen-meat-protein-ratio_100g', 'cocoa_100g', 'chlorophyl_100g',
       'carbon-footprint_100g', 'nutrition-score-fr_100g',
       'nutrition-score-uk_100g', 'glycemic-index_100g',
       'water-hardness_100g'],
      dtype='object', length=106)

# Mission 3

Crée une fonction qui prend en paramètre un dataframe, et affiche le taux de remplissage de chacune de ces colonnes sous forme de pourcentage. C'est à dire :
- colonne 1 : 73%
- colonne 2 : 99%

In [11]:
col_vide=df_yuka.notnull().sum()
#calcul du pourcentage
col_percent=round((col_vide*100)/len(df_yuka),2)
df_percent=pd.DataFrame(col_percent)

In [15]:

#définition de la fonction
def count_col(df_yuka):

    for colonne in df_yuka.columns:

#calcul du pourcentage

        col_percent=round((df_yuka.count()/len(df_yuka)*100),2)

    return col_percent

count_col(df_yuka)

code                        99.99
url                         99.99
creator                    100.00
created_t                  100.00
created_datetime           100.00
                            ...  
carbon-footprint_100g        0.08
nutrition-score-fr_100g     68.96
nutrition-score-uk_100g     68.96
glycemic-index_100g          0.00
water-hardness_100g          0.00
Length: 162, dtype: float64

In [11]:
col_percent

NameError: name 'col_percent' is not defined

# Mission 4

Quel est le nombre de colonne appartenant à chaque quartile des taux de remplissage ? Autrement dit, combien de colonne sont remplies à 0%, entre 0 et 25%, entre 25 et 50%...

In [16]:
df_yuka.notnull().mean() * 100

code                       99.992830
url                        99.992830
creator                    99.999377
created_t                  99.999065
created_datetime           99.997194
                             ...    
carbon-footprint_100g       0.083548
nutrition-score-fr_100g    68.961755
nutrition-score-uk_100g    68.961755
glycemic-index_100g         0.000000
water-hardness_100g         0.000000
Length: 162, dtype: float64

In [19]:
dico_quartile = {
    '0' : 0,
    '0-25' : 0,
    '25-50' : 0,
    '50-75' : 0,
    '75-100' : 0
}

for colonne in df_yuka.columns:
  tr = df_yuka[colonne].notnull().mean() * 100
  if tr == 0:
    dico_quartile['0'] += 1
  elif 0 < tr <= 25:
    dico_quartile['0-25'] += 1
  elif 25 < tr <= 50:
    dico_quartile['25-50'] += 1
  elif 50 < tr <= 75:
    dico_quartile['50-75'] += 1
  else:
    dico_quartile['75-100'] += 1

dico_quartile

{'0': 16, '0-25': 96, '25-50': 16, '50-75': 6, '75-100': 28}

# Mission 5

Nous voulons nous concentrer sur les aliments disponibles en France, là ou nous allons lancer la plateforme. Pour cela sélectionner uniquement les lignes dont la valeur du countries_fr contient le mot "France".

In [2]:
df_yuka_fr=df_yuka[df_yuka['countries_fr'].str.contains("France", na=False)]
df_yuka_fr

code                                                url  \
0                3087  http://world-fr.openfoodfacts.org/produit/0000...   
46              24600  http://world-fr.openfoodfacts.org/produit/0000...   
48              27205  http://world-fr.openfoodfacts.org/produit/0000...   
106             36252  http://world-fr.openfoodfacts.org/produit/0000...   
136             39259  http://world-fr.openfoodfacts.org/produit/0000...   
...               ...                                                ...   
320761  9906410000009  http://world-fr.openfoodfacts.org/produit/9906...   
320763       99111250  http://world-fr.openfoodfacts.org/produit/9911...   
320764           9918  http://world-fr.openfoodfacts.org/produit/9918...   
320765  9935010000003  http://world-fr.openfoodfacts.org/produit/9935...   
320770  9980282863788  http://world-fr.openfoodfacts.org/produit/9980...   

                           creator   created_t      created_datetime  \
0       openfoodfacts-contributors  1474103866  2016-09-17T09:17:46Z   
46                 date-limite-app  1434530704  2015-06-17T08:45:04Z   
48                         tacinte  1458238630  2016-03-17T18:17:10Z   
106                        tacinte  1422221701  2015-01-25T21:35:01Z   
136                        tacinte  1422221773  2015-01-25T21:36:13Z   
...                            ...         ...                   ...   
320761                  agamitsudo  1373480408  2013-07-10T18:20:08Z   
320763                    balooval  1367163039  2013-04-28T15:30:39Z   
320764                  woshilapin  1430167954  2015-04-27T20:52:34Z   
320765                    sebleouf  1446293229  2015-10-31T12:07:09Z   
320770  openfoodfacts-contributors  1492340089  2017-04-16T10:54:49Z   

       last_modified_t last_modified_datetime                product_name  \
0           1474103893   2016-09-17T09:18:13Z          Farine de blé noir   
46          1434535914   2015-06-17T10:11:54Z               Filet de bœuf   
48          1458238638   2016-03-17T18:17:18Z                         NaN   
106         1489055667   2017-03-09T10:34:27Z              Lion Peanut x2   
136         1473538082   2016-09-10T20:08:02Z                     Twix x2   
...                ...                    ...                         ...   
320761      1451851215   2016-01-03T20:00:15Z   Roussette du Bugey (2011)   
320763      1371690556   2013-06-20T01:09:16Z          Thé vert Earl grey   
320764      1430167992   2015-04-27T20:53:12Z  Cheese cake thé vert, yuzu   
320765      1446376839   2015-11-01T11:20:39Z              Rillette d'oie   
320770      1492340089   2017-04-16T10:54:49Z    Tomates aux Vermicelles    

               generic_name quantity  ... ph_100g fruits-vegetables-nuts_100g  \
0                       NaN      1kg  ...     NaN                         NaN   
46                      NaN  2.46 kg  ...     NaN                         NaN   
48                      NaN      NaN  ...     NaN                         NaN   
106                     NaN      NaN  ...     NaN                         NaN   
136                     NaN      NaN  ...     NaN                         NaN   
...                     ...      ...  ...     ...                         ...   
320761  Vins blanc du Bugey   750 ml  ...     NaN                         NaN   
320763    thé bio équitable     50 g  ...     NaN                         NaN   
320764                  NaN      NaN  ...     NaN                         NaN   
320765                  NaN    180 g  ...     NaN                         NaN   
320770                  NaN      67g  ...     NaN                         NaN   

       collagen-meat-protein-ratio_100g cocoa_100g chlorophyl_100g  \
0                                   NaN        NaN             NaN   
46                                  NaN        NaN             NaN   
48                                  NaN        NaN             NaN   
106                                 NaN        NaN           

# Mission 6

Un dataset trop pauvre en data n'est pas vraiment intéressant. Après avoir analyser les colonnes, nous nous sommes rendu compte que nous pouvons supprimer les colonnes avec moins de 60% de valeurs non nulles, sauf les colonnes pnns_groups_1 et pnns_groups_2, nutrition-score-fr_100g et nutrition-score-uk_100g. En effet ces 2 colonnes vont nous être utiles par la suite.

Si ce n'est pas déjà fait, supprime également les colonnes suivantes car elles ne sont pas vraiment utiles pour répondre à notre problématique :
* creator
* created_t
* created_datetime
* last_modified_t
* brands_tags
* countries
* countries_tags
* additives_tags
* states
* states_tags

In [67]:
df_yuka_fr

code                                                url  \
0                3087  http://world-fr.openfoodfacts.org/produit/0000...   
46              24600  http://world-fr.openfoodfacts.org/produit/0000...   
48              27205  http://world-fr.openfoodfacts.org/produit/0000...   
136             39259  http://world-fr.openfoodfacts.org/produit/0000...   
182           5200016  http://world-fr.openfoodfacts.org/produit/0000...   
...               ...                                                ...   
320761  9906410000009  http://world-fr.openfoodfacts.org/produit/9906...   
320763       99111250  http://world-fr.openfoodfacts.org/produit/9911...   
320764           9918  http://world-fr.openfoodfacts.org/produit/9918...   
320765  9935010000003  http://world-fr.openfoodfacts.org/produit/9935...   
320770  9980282863788  http://world-fr.openfoodfacts.org/produit/9980...   

                           creator   created_t      created_datetime  \
0       openfoodfacts-contributors  1474103866  2016-09-17T09:17:46Z   
46                 date-limite-app  1434530704  2015-06-17T08:45:04Z   
48                         tacinte  1458238630  2016-03-17T18:17:10Z   
136                        tacinte  1422221773  2015-01-25T21:36:13Z   
182                        sigoise  1441186657  2015-09-02T09:37:37Z   
...                            ...         ...                   ...   
320761                  agamitsudo  1373480408  2013-07-10T18:20:08Z   
320763                    balooval  1367163039  2013-04-28T15:30:39Z   
320764                  woshilapin  1430167954  2015-04-27T20:52:34Z   
320765                    sebleouf  1446293229  2015-10-31T12:07:09Z   
320770  openfoodfacts-contributors  1492340089  2017-04-16T10:54:49Z   

       last_modified_t last_modified_datetime                product_name  \
0           1474103893   2016-09-17T09:18:13Z          Farine de blé noir   
46          1434535914   2015-06-17T10:11:54Z               Filet de bœuf   
48          1458238638   2016-03-17T18:17:18Z                         NaN   
136         1473538082   2016-09-10T20:08:02Z                     Twix x2   
182         1442570752   2015-09-18T10:05:52Z           lentilles vertes    
...                ...                    ...                         ...   
320761      1451851215   2016-01-03T20:00:15Z   Roussette du Bugey (2011)   
320763      1371690556   2013-06-20T01:09:16Z          Thé vert Earl grey   
320764      1430167992   2015-04-27T20:53:12Z  Cheese cake thé vert, yuzu   
320765      1446376839   2015-11-01T11:20:39Z              Rillette d'oie   
320770      1492340089   2017-04-16T10:54:49Z    Tomates aux Vermicelles    

               generic_name quantity  ... ph_100g fruits-vegetables-nuts_100g  \
0                       NaN      1kg  ...     NaN                         NaN   
46                      NaN  2.46 kg  ...     NaN                         NaN   
48                      NaN      NaN  ...     NaN                         NaN   
136                     NaN      NaN  ...     NaN                         NaN   
182                     NaN    1 kg   ...     NaN                         NaN   
...                     ...      ...  ...     ...                         ...   
320761  Vins blanc du Bugey   750 ml  ...     NaN                         NaN   
320763    thé bio équitable     50 g  ...     NaN                         NaN   
320764                  NaN      NaN  ...     NaN                         NaN   
320765                  NaN    180 g  ...     NaN                         NaN   
320770                  NaN      67g  ...     NaN                         NaN   

       collagen-meat-protein-ratio_100g cocoa_100g chlorophyl_100g  \
0                                   NaN        NaN             NaN   
46                                  NaN        NaN             NaN   
48                                  NaN        NaN             NaN   
136                                 NaN        NaN           

In [3]:
df_yuka_fr_colagarder=['pnns_groups_1' , 'pnns_groups_2', 'nutrition-score-fr_100g' , 'nutrition-score-uk_100g']
df_yuka_fr_a_supprimer = ['creator', 'created_t', 'created_datetime', 'last_modified_t', 'brands_tags', 'countries', 'countries_tags', 'additives_tags', 'states', 'states_tags']

for colonne in df_yuka_fr.columns:
    if colonne not in df_yuka_fr_colagarder and df_yuka_fr[colonne].notnull().mean() < 0.60:
        df_yuka_fr.drop(colonne, axis=1, inplace=True)

df_yuka_fr.drop(columns=df_yuka_fr_a_supprimer, inplace=True, errors='ignore')





<ipython-input-3-9b837820eeac>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_yuka_fr.drop(colonne, axis=1, inplace=True)
<ipython-input-3-9b837820eeac>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_yuka_fr.drop(colonne, axis=1, inplace=True)
<ipython-input-3-9b837820eeac>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_yuka_fr.drop(colonne, axis=1, inplace=True)
<ipython-input-3-9b837820eeac>:6: SettingWithCopyWarning: 

In [4]:
df_yuka_fr.info()

<class 'pandas.core.frame.DataFrame'>
Index: 98440 entries, 0 to 320770
Data columns (total 24 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   code                     98440 non-null  object 
 1   url                      98440 non-null  object 
 2   last_modified_datetime   98440 non-null  object 
 3   product_name             91247 non-null  object 
 4   quantity                 71237 non-null  object 
 5   brands                   86426 non-null  object 
 6   categories               61955 non-null  object 
 7   categories_tags          61955 non-null  object 
 8   categories_fr            61955 non-null  object 
 9   countries_fr             98440 non-null  object 
 10  nutrition_grade_fr       61415 non-null  object 
 11  pnns_groups_1            64881 non-null  object 
 12  pnns_groups_2            67038 non-null  object 
 13  states_fr                98440 non-null  object 
 14  main_category            6

# Mission 7

Exporte ton travail au format csv



In [5]:
df_yuka_fr.to_csv('yuka_fr.csv',index=False)